docker-multinode/cni-plugin.sh

```diff
@@ -0,0 +1,235 @@
+
+
+# Start etcd on the master node
+kube::multinode::start_etcd() {
+
+  kube::log::status "Launching etcd..."
+  
+  docker run -d \
+    --restart=${RESTART_POLICY} \
+    --net=host \
+    gcr.io/google_containers/etcd-${ARCH}:${ETCD_VERSION} \
+    /usr/local/bin/etcd \
+      --listen-client-urls=http://127.0.0.1:4001,http://${MASTER_IP}:4001 \
+      --advertise-client-urls=http://${MASTER_IP}:4001 \
+      --data-dir=/var/etcd/data
+
+  # Wait for etcd to come up
+  SECONDS=0
+  while [[ $(curl -fs http://localhost:4001/v2/machines 2>&1 1>/dev/null; echo $?) != 0 ]]; do
+    if [[ ${SECONDS} == ${TIMEOUT_FOR_SERVICES} ]]; then
+      kube::log::error "etcd failed to start. Exiting..."
+      exit
+    fi
+    sleep 1
+  done
+
+  # Set flannel net config
+  docker run \
+      --net=host \
+      gcr.io/google_containers/etcd-${ARCH}:${ETCD_VERSION} \
+      etcdctl \
+      set /coreos.com/network/config \
+          "{ \"Network\": \"${FLANNEL_NETWORK}\", \"Backend\": {\"Type\": \"${FLANNEL_BACKEND}\"}}"
+
+  sleep 2
+}
+
+# Start flannel
+kube::multinode::start_flannel() {
+
+  kube::log::status "Launching flannel..."
+
+  docker run -d \
+    --restart=${RESTART_POLICY} \
+    --net=host \
+    --privileged \
+    -v /dev/net:/dev/net \
+    -v ${FLANNEL_SUBNET_DIR}:${FLANNEL_SUBNET_DIR} \
+    gcr.io/google_containers/flannel-${ARCH}:${FLANNEL_VERSION} \
+    /opt/bin/flanneld \
+      --etcd-endpoints=http://${MASTER_IP}:4001 \
+      --ip-masq="${FLANNEL_IPMASQ}" \
+      --iface="${NET_INTERFACE}"
+
+  # Wait for the flannel subnet.env file to be created instead of a timeout. This is faster and more reliable
+  SECONDS=0
+  while [[ ! -f ${FLANNEL_SUBNET_DIR}/subnet.env ]]; do
+    if [[ ${SECONDS} == ${TIMEOUT_FOR_SERVICES} ]]; then
+      kube::log::error "flannel failed to start. Exiting..."
+      exit
+    fi
+    sleep 1
+  done
+
+  source ${FLANNEL_SUBNET_DIR}/subnet.env
+
+  kube::log::status "FLANNEL_SUBNET is set to: ${FLANNEL_SUBNET}"
+  kube::log::status "FLANNEL_MTU is set to: ${FLANNEL_MTU}"
+}
+
+# Configure docker, then restart it
+kube::multinode::restart_docker(){
+
+  case "${lsb_dist}" in
+    amzn)
+      DOCKER_CONF="/etc/sysconfig/docker"
+      sed -i.bak 's/^\(MountFlags=\).*/\1shared/' $DOCKER_CONF
+      service docker restart
+      ;;
+    centos)
+      
+      # Newer centos releases uses systemd. Handle that
+      if kube::helpers::command_exists systemctl; then
+        kube::multinode::restart_docker_systemd
+      else
+        DOCKER_CONF="/etc/sysconfig/docker"
+        sed -i.bak 's/^\(MountFlags=\).*/\1shared/' $DOCKER_CONF
+        systemctl restart docker
+      fi
+      ;;
+    ubuntu|debian)
+      # Newer ubuntu and debian releases uses systemd. Handle that
+      if kube::helpers::command_exists systemctl; then
+        kube::multinode::restart_docker_systemd
+      else
+        DOCKER_CONF="/etc/default/docker"
+        sed -i.bak 's/^\(MountFlags=\).*/\1shared/' $DOCKER_CONF
```

Comment:
Setting `shared` for this flag shows clear intention and purpose and doesn't require any comments. When we get rid of this flag then it can be unclear for others and we have to add additional comments.